features/bsm/service/impl/src/main/java/org/opennms/netmgt/bsm/service/internal/BusinessServiceManagerImpl.java

```diff
@@ -125,33 +126,17 @@ public boolean removeIpInterface(Long serviceId, Integer ipServiceId) {
     }
 
     @Override
-    public OnmsSeverity calculateStatus(Long serviceId) {
+    public OnmsSeverity getOperationalStatusForBusinessService(Long serviceId) {
         final BusinessService service = getBusinessService(serviceId);
-        final List<BusinessStatusEntity> statusEntities = getDao().getStatus();
-        if (!statusEntities.isEmpty()) {
-            final Set<BusinessStatusEntity.Key> keySet = new HashSet<>();
-            for (OnmsMonitoredService eachMonitoredService : service.getIpServices()) {
-                BusinessStatusEntity.Key key = new BusinessStatusEntity.Key(eachMonitoredService.getNodeId(), eachMonitoredService.getServiceId(), eachMonitoredService.getIpAddress());
-                keySet.add(key);
-            }
+        final OnmsSeverity severity = businessServiceStateMachine.getOperationalStatus(service);
+        return severity != null ? severity : OnmsSeverity.INDETERMINATE;
+    }
```

Comment:
I guess, this should return OnmsSeverity.NORMAL instead of INDETERMINATE, otherwise the IT will fail (see below).